File: Day 3/problem1/C_Kill_the_Monster.py
import cmath
import math
def solve():
    ch = [int(x) for x in input().split()]
    mo = [int(x) for x in input().split()]
    k,*up = (int(x) for x in input().split())
    up[0],up[1] = up[1],up[0]
    i = 0
    while i <= k:
        ch[0] += up[0]*i
        ch[1] += up[1]*(k-i)
        pow_c = math.ceil(ch[0]/mo[1])
        pow_m = math.ceil(mo[0]/ch[1])
        ch[0] -= up[0]*i
        ch[1] -= up[1]*(k-i)
        if(pow_c >= pow_m):
            print("YES")
            return
        i += 1
    print("NO")
    
t = int(input())
while t:
    solve()
    t -= 1

File: Day 3/problem1/C_Kill_the_Monster.cpp
                                      /*بسم الله الرحمن الرحيم*/       
#include <iostream>
#include <utility>
using namespace std;
#define Abd_ElQader_ ios_base::sync_with_stdio(false),cin.tie(NULL),cout.tie(NULL);
#define dbg(v) cerr << #v << " = " << v << endl;
#define vdbg(v)cerr << #v << ":\n";\
        for(int i = 0 ; i < len(v) ; i++) { cerr << v[i] << " \n" [i==len(v)-1]; }
#define len(v) (int)v.size()
#define Ceil(a,b) ((ll)a+(ll)(b-1))/b
#define h first
#define d second
typedef long long ll;

void file_name(string s) {
    freopen((s+".in").c_str(),"r",stdin);
    freopen((s+".out").c_str(),"w",stdout);
}

void solve() {
    pair<ll, ll>ch;
    pair<ll, ll>mo;
    ll k;
    pair<ll,ll>up;
    cin >> ch.h >> ch.d;
    cin >> mo.h >> mo.d;
    cin >> k;
    cin >> up.d >> up.h;
    for (int i = 0; i <= k; i++) {
        ch.h += up.h*i;
        ch.d += up.d*(k-i);
        ll pow_c = Ceil(ch.h,mo.d);
        ll pow_m = Ceil(mo.h,ch.d);
        ch.h -= up.h*i;
        ch.d -= up.d*(k-i);

        if(pow_c >= pow_m) {
            cout << "YES\n";
            return;
        }
    }
    cout << "NO\n";
}  

signed main() {
    Abd_ElQader_

    #ifndef ONLINE_JUDGE
    file_name("test");
    #endif

    int t=1;
    cin>>t;
    while(t--) 
    solve();
}
